llvm/lib/CodeGen/MIRParser/MIRParser.cpp

```diff
@@ -703,7 +703,7 @@ bool MIRParserImpl::parseRegisterInfo(PerFunctionMIParsingState &PFS,
         return error(FlagStringValue.SourceRange.Start,
                      Twine("use of undefined register flag '") +
                          FlagStringValue.Value + "'");
-      Info.Flags.push_back(FlagValue);
+      Info.Flags |= FlagValue;
```

Comment:
We should also catch if a flag is set multiple times

Reply:
Would a warning be better for it?

Reply:
No reason to be permissive

Reply:
Added the check but can't be tested till relanding the WWM_REG flag patch.